kong/pdk/vault.lua

```diff
@@ -1274,7 +1283,21 @@ local function new(self)
     -- we should refresh the secret at this point
     local ok, err = get_from_vault(reference, strategy, config, new_cache_key, parsed_reference)
```

Comment:
> I wonder if we really want this. There are _certainly_ cases where a vault backend is down for a prolonged period in time. As a user I would expect things to continue working when my backend comes back up, right?
> 
> I aggree that we need to have an upper limit of sorts but that approach seems too drastic. What do others think?

I had a similar thought. Also, maybe a setting that allows defining an upper bound in time instead of number of tries would be more user friendly (set an optional maximum downtime that is acceptable), but I guess that'd make the implementation more complex.

Would the error returned on this line give any information that helps determining if the secret was **deleted** or the vault was down / did not respond? If so maybe we could only remove the entry when we know it was deleted.

Reply:
The error message of `the secret was deleted` depends on the implementation of the secret manager products, so  very hard to tell...

Reply:
personally I like the time-based upper bound solution. We can track `last_seen` and compare it with the defined upper limit (which should be configurable).

Reply:
Alright, let me put up another PR(that implement time-based upper bond) and see if that is better than this one

Reply:
@jschmid1 @samugi What do you think if we just log the error for X times(and after that not printing this error log, since it is just the error log that the user cares about), but keep the background rotating? That would both keep the rotating works even if the vault backend has been down for a long time, and not flood the user's logging system with useless errors